src/Language/PureScript/Pretty/Common.hs

```diff
@@ -149,10 +150,11 @@ prettyPrintMany f xs = do
 -- |
 -- Prints an object key, escaping reserved names.
 --
-prettyPrintObjectKey :: Text -> Text
-prettyPrintObjectKey s | s `elem` reservedPsNames = T.pack (show s)
-                       | isUnquotedKey s = s
-                       | otherwise = T.pack (show s)
+prettyPrintObjectKey :: PSString -> Text
+prettyPrintObjectKey = prettyPrintLabel . Label
+
+prettyPrintLabel :: Label -> Text
+prettyPrintLabel = T.pack . show
```

Comment:
Can we avoid `show` here? I tend to prefer monomorphic versions of functions where possible. In particular, the old code might have been forced to use `show` as that might have been the only way to render the `Text` in the desired manner, but now, `renderJSON` on the underlying `PSString` ought to do the right thing, right?